### ISSUE TITLE
Compatibility issues with other bundles like Media while using composer update

### ISSUE THREAD

**Issue by enekochan** [sonata-project/SonataClassificationBundle#187]
I created a Symfony project (2.7) almost a year ago with SonataAdminBundle, SonataMediaBundle, SonataClassificationBundle and all their dependencies. I used "dev-master" at the time for most of them and everything was installed with composer and SonataEasyExtendsBundle flawlessly. I used the "dev-master" branched because I needed the `Context` entity feature and also wanted the latest SonataAdmin (with the new UI and such). Now I'm trying to update that very same project with `composer update` and always, no matter what, I end up having dependency issues. It's like if I had a deadlock with the different versions that work well together and the versions I need (the mentioned Context for example): Block needs this version of Core but Media needs this other one, ...

The best I've come up with is having this error that refers to my constraint:

```
  Problem 1
    - The requested package sonata-project/classification-bundle ~2.3 exists as sonata-project/classification-bundle[2.2.0, 2.2.1, 2.2.2, 2.2.x-dev, dev-add_context, dev-master, 3.x-dev] but these are rejected by your constraint.
  Problem 2
    - Installation request for sonata-project/media-bundle dev-master -> satisfiable by sonata-project/media-bundle[dev-master].
    - sonata-project/media-bundle dev-master requires sonata-project/classification-bundle ~2.3 -> no matching package found.
```

While trying to solve this I added those options but with no luck:

```
    "minimum-stability": "dev",
    "prefer-stable": true,
```

This is my `composer.json` right now:

```
{
    "name": "symfony/framework-standard-edition",
    "license": "MIT",
    "type": "project",
    "description": "The \"Symfony Standard Edition\" distribution",
    "autoload": {
        "psr-4": { "": "src/", "SymfonyStandard\\": "app/" }
    },
    "require": {
        "php": ">=5.4",
        "symfony/symfony": "2.7.*",
        "doctrine/orm": "dev-master",
        "doctrine/dbal": "dev-master",
        "doctrine/doctrine-bundle": "^1.6",
        "doctrine/mongodb-odm": "1.0.*@dev",
        "doctrine/mongodb-odm-bundle": "3.0.*@dev",
        "symfony/assetic-bundle": "~2.3",
        "symfony/swiftmailer-bundle": "~2.3",
        "symfony/monolog-bundle": "~2.4",
        "sensio/distribution-bundle": "~3.0,>=3.0.12",
        "sensio/framework-extra-bundle": "^3.0.13",
        "incenteev/composer-parameter-handler": "~2.0",
        "twig/extensions": "^1.3",
        "friendsofsymfony/user-bundle": "~2.0@dev",
        "friendsofsymfony/jsrouting-bundle": "^1.5",
        "friendsofsymfony/rest-bundle": "dev-master",
        "nelmio/api-doc-bundle": "^2.11",
        "jms/serializer-bundle": "~0.11",
        "lexik/jwt-authentication-bundle": "^1.4",
        "gesdinet/jwt-refresh-token-bundle": "^0.1.4",
        "hwi/oauth-bundle": "^0.3.9",
        "willdurand/js-translation-bundle": "dev-master",
        "willdurand/geocoder-bundle": "^4.0",
        "willdurand/negotiation": "^2.0",

        "sonata-project/core-bundle": "dev-master",
        "sonata-project/admin-bundle": "dev-master",
        "sonata-project/doctrine-orm-admin-bundle": "dev-master",
        "sonata-project/block-bundle": "dev-master",
        "sonata-project/easy-extends-bundle": "^2.1",
        "sonata-project/intl-bundle": "dev-master",
        "sonata-project/datagrid-bundle": "dev-master",
        "sonata-project/classification-bundle": "dev-master",
        "sonata-project/media-bundle": "dev-master",

        "stof/doctrine-extensions-bundle": "^1.2",
        "egeloen/ckeditor-bundle": "~2.0",
        "egeloen/google-map-bundle": "^2.2",
        "oneup/uploader-bundle": "~1.4",
        "ddeboer/data-import": "@stable",
        "ddeboer/data-import-bundle": "~0.1",
        "phpoffice/phpexcel": "^1.8",
        "bushidoio/pdf-bundle": "dev-master",
        "messagebird/php-rest-api": "1.2.x",
        "facebook/php-sdk-v4": "^5.1",
        "fzaninotto/faker": "~1.2",
        "raulfraile/ladybug-bundle": "~1.0"
    },
    "require-dev": {
        "sensio/generator-bundle": "~2.3",
        "doctrine/doctrine-fixtures-bundle": "dev-master"
    },
    "scripts": {
        "post-root-package-install": [
            "SymfonyStandard\\Composer::hookRootPackageInstall"
        ],
        "post-install-cmd": [
            "Incenteev\\ParameterHandler\\ScriptHandler::buildParameters",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::buildBootstrap",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::clearCache",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installAssets",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installRequirementsFile",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::removeSymfonyStandardFiles",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::prepareDeploymentTarget"
        ],
        "post-update-cmd": [
            "Incenteev\\ParameterHandler\\ScriptHandler::buildParameters",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::buildBootstrap",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::clearCache",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installAssets",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installRequirementsFile",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::removeSymfonyStandardFiles",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::prepareDeploymentTarget"
        ]
    },
    "config": {
        "bin-dir": "bin"
    },
    "extra": {
        "symfony-app-dir": "app",
        "symfony-web-dir": "web",
        "symfony-assets-install": "relative",
        "incenteev-parameters": {
            "file": "app/config/parameters.yml"
        },
        "branch-alias": {
            "dev-master": "2.7-dev"
        }
    }
}
```

Have tried the versions for the bundles I need from the `composer.json` in [Sonata Sandbox 2-4-dev](https://github.com/sonata-project/sandbox/blob/2.4-develop/composer.json) but I get this error:

```
  Problem 1
    - Installation request for sonata-project/core-bundle 2.3.x-dev -> satisfiable by sonata-project/core-bundle[2.3.x-dev].
    - sonata-project/doctrine-orm-admin-bundle dev-master requires sonata-project/core-bundle 3.x-dev@dev -> satisfiable by sonata-project/core-bundle[3.x-dev].
    - remove sonata-project/doctrine-orm-admin-bundle 2.4.x-dev|keep sonata-project/doctrine-orm-admin-bundle dev-master
    - Can only install one of: sonata-project/core-bundle[3.x-dev, 2.3.x-dev].
    - Installation request for sonata-project/doctrine-orm-admin-bundle ~2.4@dev -> satisfiable by sonata-project/doctrine-orm-admin-bundle[2.4.x-dev].
```

I'm also trying to create a new project like I did before but I've only been able to skip this problem with ClassificationBundle/MediaBundle using the versions in `composer.json` bellow, the problem is that with those I don't have the latest version of SonataAdmin neither the Context entity for SonataClassificationBundle:

```
        "sonata-project/core-bundle": "2.x-dev",
        "sonata-project/admin-bundle": "2.3.x-dev",
        "sonata-project/doctrine-orm-admin-bundle": "2.3.x-dev",
        "sonata-project/block-bundle": "2.2.x-dev",
        "sonata-project/easy-extends-bundle": "^2.1",
        "sonata-project/intl-bundle": "dev-master",
        "sonata-project/datagrid-bundle": "dev-master",
        "sonata-project/classification-bundle": "2.2.*",
        "sonata-project/media-bundle": "2.3.*",
```

**Comment by enekochan** [comment]
Everything fixed using the new 3.0 versions:

```
        "sonata-project/core-bundle": "^3.0",
        "sonata-project/admin-bundle": "^3.0",
        "sonata-project/doctrine-orm-admin-bundle": "^3.0",
        "sonata-project/block-bundle": "^3.0",
        "sonata-project/easy-extends-bundle": "^2.1",
        "sonata-project/intl-bundle": "^2.2",
        "sonata-project/datagrid-bundle": "^2.2",
        "sonata-project/classification-bundle": "^3.0",
        "sonata-project/media-bundle": "^3.0",
```

**Comment by OskarStark** [comment]
Thanks for the feedback @enekochan 👍
